packages/web3-eth/src/schemas.ts

```diff
@@ -0,0 +1,386 @@
+			type: 'array',
+			items: {
+				eth: 'bytes32',
+			},
+		},
+	},
+};
+
+export const accessListSchema = {
+	type: 'array',
+	items: {
+		...accessListItemSchema,
+	},
+};
+
+export const chainSchema = {
+	type: 'string',
+	enum: ['goerli', 'kovan', 'mainnet', 'rinkeby', 'ropsten', 'sepolia'],
+};
+
+export const hardforkSchema = {
+	type: 'string',
+	enum: [
+		'arrowGlacier',
+		'berlin',
+		'byzantium',
+		'chainstart',
+		'constantinople',
+		'dao',
+		'homestead',
+		'istanbul',
+		'london',
+		'merge',
+		'muirGlacier',
+		'petersburg',
+		'shanghai',
+		'spuriousDragon',
+		'tangerineWhistle',
+	],
+};
+
+export const customChainSchema = {
+	type: 'object',
+	properties: {
+		name: {
+			eth: 'string',
+		},
+		networkId: {
+			eth: 'uint',
+		},
+		chainId: {
+			eth: 'uint',
+		},
+	},
+};
+
+export const transactionSchema = {
+	type: 'object',
+	properties: {
+		from: {
+			eth: 'address',
+		},
+		to: {
+			eth: 'address',
+		},
+		value: {
+			eth: 'uint',
+		},
+		gas: {
+			eth: 'uint',
+		},
+		gasPrice: {
+			eth: 'uint',
+		},
+		type: {
+			eth: 'uint',
+		},
+		maxFeePerGas: {
+			eth: 'uint',
+		},
+		maxPriorityFeePerGas: {
+			eth: 'uint',
+		},
+		accessList: {
+			...accessListSchema,
+		},
+		data: {
+			eth: 'bytes',
+		},
+		input: {
+			eth: 'bytes',
+		},
+		nonce: {
+			eth: 'uint',
+		},
+		chain: { ...chainSchema },
+		hardfork: { ...hardforkSchema },
+		chainId: {
+			eth: 'uint',
+		},
+		networkId: {
+			eth: 'uint',
+		},
+		common: {
+			type: 'object',
+			properties: {
+				customChain: { ...customChainSchema },
+				baseChain: {
+					...chainSchema,
+				},
+				hardfork: {
+					...hardforkSchema,
+				},
+			},
+		},
+		gasLimit: {
+			eth: 'uint',
+		},
+		v: {
+			eth: 'uint',
+		},
+		r: {
+			eth: 'bytes',
+		},
+		s: {
+			eth: 'bytes',
+		},
+	},
+};
+
+export const transactionInfoSchema = {
+	type: 'object',
+	properties: {
+		blockHash: {
+			eth: 'bytes32',
+		},
+		blockNumber: {
+			eth: 'uint',
+		},
+		from: {
+			eth: 'address',
+		},
+		hash: {
+			eth: 'bytes32',
+		},
+		transactionIndex: {
+			eth: 'uint',
+		},
+	},
+};
+
+export const blockSchema = {
+	type: 'object',
+	properties: {
+		parentHash: {
+			eth: 'bytes32',
+		},
+		sha3Uncles: {
+			eth: 'bytes32',
+		},
+		miner: {
+			eth: 'bytes',
+		},
+		stateRoot: {
+			eth: 'bytes32',
+		},
+		transactionsRoot: {
+			eth: 'bytes32',
+		},
+		receiptsRoot: {
+			eth: 'bytes32',
+		},
+		logsBloom: {
+			eth: 'bytes256',
+		},
+		difficulty: {
+			eth: 'uint',
+		},
+		number: {
+			eth: 'uint',
+		},
+		gasLimit: {
+			eth: 'uint',
+		},
+		gasUsed: {
+			eth: 'uint',
+		},
+		timestamp: {
+			eth: 'uint',
+		},
+		extraData: {
+			eth: 'bytes',
+		},
+		mixHash: {
+			eth: 'bytes32',
+		},
+		nonce: {
+			eth: 'uint',
+		},
+		totalDifficulty: {
+			eth: 'uint',
+		},
+		baseFeePerGas: {
+			eth: 'uint',
+		},
+		size: {
+			eth: 'uint',
+		},
+		// TODO: This attribute can be array of hashes or transaction info
```

Comment:
Do the schemas not currently support multiple types for properties like this? How difficult is this to figure out?

Reply:
In JSON-Schema, we use `anyOf` keyword for adding such conditional schema. The formatter does not support it yet, but we can add this support in future.